pom.xml

```diff
@@ -469,6 +473,13 @@
             <optional>true</optional>
         </dependency>
 
+        <!-- TOTP client -->
+        <dependency>
+            <groupId>com.warrenstrange</groupId>
+            <artifactId>googleauth</artifactId>
+            <version>1.1.2</version>
```

Comment:
Maybe exclude `org.apache.httpcomponents:httpclient` here? This shouldn't break anything, because we don't use the web API adapter.

Reply:
`commons-codec:commons-codec` (for Base32 and Base64 encoding/decoding) is also present in Minecraft since 1.7.10, but in a different version (1.9 vs 1.10) and therefore could maybe break something.

Reply:
Good point.

Reply:
As an update, looks like httpclient is required when generating the QR code with Google :( https://postimg.org/image/68bv03bz1/

Reply:
What's your suggestion for the commons-codec thing? Should we really exclude it? This would mean that for Sponge etc. we'd also rely on it being there for everything to work.

Reply:
>As an update, looks like httpclient is required when generating the QR code with Google :( https://postimg.org/image/68bv03bz1/

Okay, my fault sorry. I always generated the URLs on my own. It makes more sense to use the provided method than creating something custom. So looks like we have to shade `httpclient` and `httpcore` then. 

>This would mean that for Sponge etc. we'd also rely on it being there for everything to work.

Good point. 

Base32 and Base64 encoding is also available in guava which we already shade into the final jar, but it would be ridiculous to fork every project that includes dependencies we do not have.

So yeah, then we have to shade it too.